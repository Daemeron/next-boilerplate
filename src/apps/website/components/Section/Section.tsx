import React, { ReactChild } from 'react';
import styled from 'styled-components';
import { ThemePropsInterface } from '@website/core';

interface SectionProps {
  children?: ReactChild | ReactChild[];
}

export const Section = (props: SectionProps): JSX.Element => <SectionBox>{props.children}</SectionBox>;

export const SectionBox = styled.div<ThemePropsInterface>`
  display: flex;
  justify-content: center;
  padding: 1rem 0;

  @media (min-width: 768px) {
    padding: 2rem 0;
  }
  @media (min-width: 992px) {
    padding: 3rem 0;
  }
`;
