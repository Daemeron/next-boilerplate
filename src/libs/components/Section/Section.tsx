import React, { ReactChild } from 'react';
import styled from 'styled-components';

interface SectionProps {
  children?: ReactChild | ReactChild[];
}

export const Section = (props: SectionProps): JSX.Element => <SectionBox>{props.children}</SectionBox>;

export const SectionBox = styled.div``;
