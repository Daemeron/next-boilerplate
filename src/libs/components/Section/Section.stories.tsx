import React, { Fragment } from 'react';

import { Section } from './Section';

export default {
  title: 'Components/Section',
};

export const WithText = (): JSX.Element => (
  <Fragment>
    <Section>dummy text</Section>
  </Fragment>
);
