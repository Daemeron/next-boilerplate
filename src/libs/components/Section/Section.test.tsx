import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Section } from './Section';

describe('Section', () => {
  test('matches snapshot', () => {
    const section = render(<Section>test</Section>);
    expect(section.container).toMatchSnapshot();
  });

  test('renders single child', () => {
    const section = render(<Section>test</Section>);
    expect(section.queryByText('test')).toBeTruthy();
  });

  test('renders multiple children', () => {
    const section = render(
      <Section>
        <p>test</p>
        <p>test</p>
      </Section>
    );
    expect(section.queryAllByText('test').length).toEqual(2);
  });
});
