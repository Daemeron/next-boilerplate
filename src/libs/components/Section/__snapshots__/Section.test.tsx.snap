// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `
<div>
  <div
    class=""
  >
    test
  </div>
</div>
`;
